packages/create-yoshi-app/templates/business-manager-module/typescript/src/client.tsx

```diff
@@ -1,29 +1,42 @@
 import axios from 'axios';
-import React from 'react';
-import { I18nextProvider } from 'react-i18next';
+import React, { FC, useRef, useMemo } from 'react';
 import { notifyViewStartLoading } from '@wix/business-manager-api';
 import { wixAxiosConfig } from '@wix/wix-axios-config';
+import { I18nextProvider, initI18n } from '@wix/wix-i18n-config';
 import { COMPONENT_NAME, IBMModuleParams } from './config';
-import i18n from './i18n';
 import App from './components/App';
 
 wixAxiosConfig(axios, {
   baseURL: '/',
 });
 
-export default class AppContainer extends React.Component<IBMModuleParams> {
-  constructor(props: IBMModuleParams) {
-    super(props);
-    notifyViewStartLoading(COMPONENT_NAME);
-  }
+const useOnce = (cb: () => void) => {
+  const ref = useRef(false);
 
-  render() {
-    const { locale, config } = this.props;
-    const baseUrl = config.topology.staticsUrl;
-    return (
-      <I18nextProvider i18n={i18n(locale, baseUrl)}>
-        <App />
-      </I18nextProvider>
-    );
+  if (!ref.current) {
+    ref.current = true;
+    cb();
   }
-}
+};
+
+const AppContainer: FC<IBMModuleParams> = ({ locale }) => {
+  useOnce(() => notifyViewStartLoading(COMPONENT_NAME));
+
+  const i18n = useMemo(
+    () =>
+      initI18n({
+        locale,
+        asyncMessagesLoader: locale2 =>
```

Comment:
is locale2 really required? Can't is just be:
```suggestion
        asyncMessagesLoader: () => import(`./assets/locale/messages_${locale}.json`),
```

Reply:
Yeah this made me think, if we have to explicitly pass `locale` on each "init" call, why must the loader function accept a locale?
Is there a different use case where we don't pass `locale`?
I'll fix our usage regardless :)

Reply:
In other cases i18n is initialized at file/module level, rather than component. So if you change locale after app has been initialized, you'll need a loader function to know what locale to load. That's why it accepts argument.